src/scgpt/embedding/test.py

```diff
@@ -0,0 +1,262 @@
+        "--model_config", model_config_file,
+        "--dsbn", "True",
+        "--obsm_gene_tokens", "gene_id_tokens",
+        "--obsm_tokenized_values", "values_tokenized",
+        "--obsm_padding_mask", "padding_mask",
+        "--output", output_embedding_file
+    ]
+
+    with pytest.raises(subprocess.CalledProcessError) as err:
+        run_component(args)
+    assert re.search(
+        r"ValueError: When dsbn is set to True, you are required to provide batch labels \(input_obs_batch_labels\)\.",
+        err.value.stdout.decode('utf-8'))
+
+def test_integration_embedding_non_existing_keys(run_component, tmp_path):
+    output_embedding_file = tmp_path / "Kim2020_Lung_subset_embedded.h5mu"
+
+    # Test for non-existing gene names key
+    args_1 = [
+        "--input", tokenized_data_path,
+        "--modality", "rna",
+        "--model", model_file,
+        "--model_vocab", vocab_file,
+        "--model_config", model_config_file,
+        "--dsbn", "True",
+        "--obs_batch_label", "sample",
+        "--var_gene_names", "dummy_gene_name_key",
+        "--obsm_gene_tokens", "gene_id_tokens",
+        "--obsm_tokenized_values", "values_tokenized",
+        "--obsm_padding_mask", "padding_mask",
+        "--output", output_embedding_file
+    ]
+
+    with pytest.raises(subprocess.CalledProcessError) as err:
+        run_component(args_1)
+    assert re.search(
+        r"KeyError: \'dummy_gene_name_key\'",
+        err.value.stdout.decode('utf-8'))
+
+    # Test for non-existing batch label key
+    args_2 = [
+        "--input", tokenized_data_path,
+        "--modality", "rna",
+        "--model", model_file,
+        "--model_vocab", vocab_file,
+        "--model_config", model_config_file,
+        "--dsbn", "True",
+        "--obs_batch_label", "dummy_batch_label_key",
+        "--obsm_gene_tokens", "gene_id_tokens",
+        "--obsm_tokenized_values", "values_tokenized",
+        "--obsm_padding_mask", "padding_mask",
+        "--output", output_embedding_file
+    ]
+
+    with pytest.raises(subprocess.CalledProcessError) as err:
+        run_component(args_2)
+    assert re.search(
+        r"KeyError: \'dummy_batch_label_key\'",
+        err.value.stdout.decode('utf-8'))
+
+    # Test for non-existing tokenized values key
+    args_3 = [
+        "--input", tokenized_data_path,
+        "--modality", "rna",
+        "--model", model_file,
+        "--model_vocab", vocab_file,
+        "--model_config", model_config_file,
+        "--dsbn", "True",
+        "--obs_batch_label", "sample",
+        "--obsm_gene_tokens", "gene_id_tokens",
+        "--obsm_tokenized_values", "dummy_values_tokenized",
+        "--obsm_padding_mask", "padding_mask",
+        "--output", output_embedding_file
+    ]
+
+    with pytest.raises(subprocess.CalledProcessError) as err:
+        run_component(args_3)
+    assert re.search(
+        r"KeyError: \'dummy_values_tokenized\'",
+        err.value.stdout.decode('utf-8'))
```

Comment:
Could you adjust the script so that it provides a clearer error message for this?